### PR TITLE
Docker image fixes

### PR DESCRIPTION
**Pull request by willemvd**
Docker build was broken due to different naming of the artifact from Go, paths changed and glide has been removed.
This should fix these issues (haven't got time to properly test it all)

Signed-off-by: Willem van Dreumel (willem@opencirclesolutions.nl)

### CODE REVIEW

File: docker/s6/gogs/setup [hunk 1]
```diff
@@ -21,3 +21,5 @@ ln -sf /data/git /home/git
 
 chown -R git:git /data /app/gogs ~git/
 chmod 0755 /data /data/gogs ~git/
```
Comment (bkcsoft) [inline docker/s6/gogs/setup]: shouldn't these two lines be changed as well?

Comment (bkcsoft) [reply docker/s6/gogs/setup]: And you'd have to update the Dockerfile as well then.

**Review by bkcsoft — CHANGES_REQUESTED** [review]

File: docker/s6/gogs/setup (hunk 1 above)
Comment (willemvd) [reply docker/s6/gogs/setup]: Why would this need a change?

Comment (strk) [reply docker/s6/gogs/setup]: to be named "gitea" instead of "gogs" ?

File: docker/build.sh [hunk 2]
```diff
@@ -10,20 +10,12 @@ export GO15VENDOREXPERIMENT=1
...
-mkdir -p ${GOPATH}/src/github.com/gogits/
+mkdir -p ${GOPATH}/src/github.com/go-gitea/
 ln -s /app/gogs/ ${GOPATH}/src/github.com/go-gitea/gitea
 cd ${GOPATH}/src/github.com/go-gitea/gitea
-glide install
 make build TAGS="sqlite cert pam"
+go install
 
 # Cleanup GOPATH & vendoring dir
 rm -r $GOPATH /app/gogs/vendor
```
Comment (strk) [inline docker/build.sh]: Should it be /app/gitea/vendor ?

**Comment by xinity** [comment]
i think we should move out of the Dockerfile the build process
as @tboerger will build binaries, we'll just have to add them in the shipped IMAGE.
PR incoming in my case :)

**Comment by ghost** [comment]
@xinty Actually, the whole point of a Dockerfile is to allow the user to build the binary him/herself. We should not be putting binaries in there if we can build it ourselves.

**Comment by metalmatze** [comment]
Not sure if I understand @LefsFlarey correctly. But what will happen is that everyone can simply run `make build`on their machine and create a binary. Our CI Pipeline will then use the binary created by the same command and simply copy it into the docker image. No need to have Go inside the container. Just a binary.
Right @tboerger ?

**Comment by xinity** [comment]
Agree @metalmaze

**Comment by tboerger** [comment]
It will be automated, the makefile tasks are already there, just need to add drone as ci

**Comment by willemvd** [comment]
Changed the whole docker image to be able to run it as unprivileged user (needed for some docker environments like OpenShift v3 which do not allow to run as root) and merged the current master into the pull request

**Comment by tboerger** [comment]
I'm really not a fan of the current approach. I really dislike the switch away from Alpine.

File: Dockerfile [hunk 3]
```diff
@@ -1,22 +1,22 @@
-FROM alpine:3.3
-MAINTAINER jp@roemer.im
+FROM willemvd/ubuntu-unprivileged-git-ssh:latest
```
Comment (leitzler) [inline Dockerfile:1]: This opens up a huge security hole and should not, in any way, be accepted by the maintainers of Gitea.

**Review by leitzler — CHANGES_REQUESTED** [review]

**Comment by stevenroose** [comment]
It should definitively be possible to build Gitea on an alpine container without too much hassle.

File: Dockerfile (hunk 3 above)
Comment (ghost) [reply Dockerfile:1]: I agree, what's wrong with alpine?

**Comment by tboerger** [comment]
So we agree to build only on official images

**Comment by tboerger** [comment]
Should we take https://github.com/go-gitea/gitea/pull/288 in favor of this?

**Comment by tboerger** [comment]
Thanks for the work, we have merged #288 now to properly build docker images.

**Pull request closed without merging**